feature_selection/forward_selection.py

```diff
@@ -0,0 +1,79 @@
+def forward_selection(scorer, X, y, min_features=1, max_features=5):
+    '''
+    The Forward Selection is an algorithm used to select features.
+    It starts as an empty model, and add the variable with the
+    best improvement in the model. The process is iteratively 
+    repeated and it stops when the remaining variables doesn't
+    improve the accuracy of the model.
+
+    Parameters
+    ----------
+    scorer : object
+        function given by the user that returns the score. The
+        forward_selection function, will chose the smaller score 
+        from the scorer; in other words, the scorer should be 
+        giving following the idea: "the smaller, the better"
+    X : numpy.ndarray
+        training dataset
+    y : numpy.ndarray
+        test dataset
+    min_features : int (default=None)
+        number of minimum features to select
+    max_features : int (default=10)
+        number of maximum features to select
+
+    Returns
+    -------
+    numpy ndarray
+      Numeric array of selected features
+
+    Examples
+    --------
+    >>> from sklearn.linear_model import LinearRegression
+    >>> from sklearn.datasets import make_friedman1
+    >>> data, target = make_friedman1(n_samples=200, n_features=15, random_state=0)
+    >>> def my_scorer_fn2(X, y):
+    >>>      lm = LinearRegression().fit(X, y)
+    >>>      return 1 - lm.score(X, y)
+    >>>
+    >>> forward_selection(my_scorer_fn, data, target, max_features=7)
+    array([0, 2, 3, 7, 9, 12, 13])
+    '''
+    ftr_ = []
+    no_ftr_ = []
```

Comment:
Also, avoid abbreviations in variable names when they are not obvious. In this case, for instance, I don't know what `no_ftr` means.

Reply:
Thanks for the piece of advice!